src/datasets/packaged_modules/parquet/parquet.py

```diff
@@ -63,7 +64,7 @@ def _cast_table(self, pa_table: pa.Table) -> pa.Table:
 
     def _generate_tables(self, files):
         schema = self.info.features.arrow_schema if self.info.features is not None else None
-        if self.info.features is not None and self.config.columns is not None:
+        if self.config.features is not None and self.config.columns is not None:
             if sorted(field.name for field in schema) != sorted(self.config.columns):
```

Comment:
Should the line here not check whether the configured columns are a subset of the schema columns, i.e.

```python
set(self.config.columns) <= set(field.name for field in schema)
```

Also, I'm not very familiar with the codebase, but it could be required to handle the edge case of `features` and `columns` argument being provided at the same time, in which case the features should be a subset of the requested columns.

Reply:
I think it's good to be strict here